### ISSUE TITLE
Reuse image thumbnails from UFDR reports

### ISSUE THREAD

**Issue by lfcnassif** [sepinf-inc/IPED#202]
Often image thumb creation takes 50% of processing time with large UFDRs, that is CPU intensive, even with common formats. But inside the ufdr already exists a sqlite database with image thumbnails generated by ufed. We can reuse those thumbs instead of generating them.

**Comment by tc-wleite** [comment]
@lfcnassif, this would be particularly useful for cases with many images with formats that require external processing (e.g. HEIC).

Inspecting some UFDRs that I have here (both older and recent ones) I see one downside: thumbnails stored in UFDRs have a very low resolution. It seems that there is a 100 pixels height limit.

So a 1:1 image would have a 100x100 pixels thumbnail, instead of 160x160 currently used. 
That is a drastic reduction (25600 -> 10000 pixels) which would affect the visualization quality in the gallery, and other tasks that use the thumbnail (DIE, PhotoDNA and ImageSimilarity).

**Comment by lfcnassif** [comment]
Good point! I didn't know about such low resolution used by UFDRs. Actually I was thinking about closing this as "won't fix" after #81 gave us a 3x speed up, although that just applies to basic formats. If you agree, I will close this.

**Comment by tc-wleite** [comment]
Sorry for not pointing this out earlier. 
In fact, I also only found out that UFDR thumbnails are so small recently.

About the formats that require external conversion (e.g. HEIC and WEBP), which usually take most of thumbnail generation time, I am checking from time to time new solutions that we could apply. 
I had an idea for HEIC and found a possible alternative for WEBP. 
I will try to look this subject in more detail and will create a new issue if I found a solution.

I think this one can be closed for now.
